### PR TITLE
feat: add Log4j2 plugin to filter the stacktrace

### PR DESCRIPTION
**Pull request by kayman-mk**
# Description

Adds the first version of the plugin. This is a copy of our internal repository. I adjusted the `pom.xml` and the package name.

# Checklist

- [x] My code follows the style guidelines of the project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <!-- we support Java 8 as Log4j2 does -->
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+    </properties>
+
+    <licenses>
+        <license>
+            <name>MIT License</name>
+            <url>https://github.com/Hapag-Lloyd/dist-comm-vis/blob/main/LICENSE</url>
```
Comment (PascalFrenz) [inline pom.xml:24]: Should be changed to Apache 2.0 as https://github.com/Hapag-Lloyd/log4j2-filtered-stacktrace-plugin/pull/1 changes the license too

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        <developer>
+            <organization>Hapag-Lloyd AG</organization>
+            <organizationUrl>https://www.hapag-lloyd.com/</organizationUrl>
+        </developer>
+    </developers>
+
+    <scm>
+        <connection>scm:git:git://github.com/Hapag-Lloyd/dist-comm-vis.git</connection>
+        <developerConnection>scm:git:ssh://gitHub.com:Hapag-Lloyd/dist-comm-vis.git</developerConnection>
+        <url>https://github.com/Hapag-Lloyd/dist-comm-vis/tree/main</url>
```
Comment (PascalFrenz) [inline pom.xml:38]: these links should point to this repository (https://github.com/Hapag-Lloyd/log4j2-filtered-stacktrace-plugin)

File: src/test/java/com/hlag/logging/log4j2/FilteredStacktraceStackTraceJsonResolverUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    private EventResolverContext mockedEventResolverContext;
+
+    @BeforeEach
+    void setUp() {
+        Mockito.when(mockedEventResolverContext.getRecyclerFactory()).thenReturn(new QueueingRecyclerFactory(LinkedList::new));
+        Mockito.when(mockedEventResolverContext.getMaxStringByteCount()).thenReturn(60000);
+
+        jsonWriter = JsonWriter.newBuilder().setMaxStringLength(60000).setTruncatedStringSuffix("...").build();
+
+        FilteredStacktraceExceptionResolver.setPackagesToRemoveFromStacktrace(List.of("com.hlag.fis.sales"));
```
Comment (PascalFrenz) [inline src/test/java/com/hlag/logging/log4j2/FilteredStacktraceStackTraceJsonResolverUnitTest.java:35]: change this package name to `com.hlag.logging.log4j2` or similar

**Review by PascalFrenz — CHANGES_REQUESTED** [review]

**Comment by kayman-mk** [comment]
Good spot. Everything fix. Tests still green.

File: log4j2-stacktrace-filter.iml
Comment (PascalFrenz) [inline log4j2-stacktrace-filter.iml:1]: This is an IntelliJ-specific file, committed unintentionally. I think we should add it to `.gitignore` too.

Comment (kayman-mk) [reply log4j2-stacktrace-filter.iml:1]: Just checked the file. Seems to be good to have it as it includes the module definition only.

Comment (kayman-mk) [reply log4j2-stacktrace-filter.iml:1]: Checked some other pages. LEt's get rid of the file.

**Review by PascalFrenz — APPROVED** [review]

**Pull request merged**